src/legacy/ui/public/vislib/visualizations/point_series/column_chart.js

```diff
@@ -167,17 +188,43 @@ export class ColumnChart extends PointSeries {
       if (d.y0 === 0 && yMin > 0) {
         return yScale(yMin) - yScale(d.y);
       }
-
       return Math.abs(yScale(d.y0) - yScale(d.y0 + d.y));
     }
 
+    function formatValue(d) {
+      return chartData.yAxisFormatter(d.y);
+    }
+
     // update
     bars
       .attr('x', isHorizontal ? x : y)
       .attr('width', isHorizontal ? widthFunc : heightFunc)
       .attr('y', isHorizontal ? y : x)
       .attr('height', isHorizontal ? heightFunc : widthFunc);
 
+    const layer = d3.select(bars[0].parentNode);
+    const barLabels = layer.selectAll('text').data(chartData.values.filter(function (d) {
+      return !_.isNull(d.y);
+    }));
+
+    if (isLabels) {
+      const colorFunc = this.handler.data.getColorFunc();
+      const d3Color = d3.rgb(colorFunc(chartData.label));
+      const labelColor = isColorDark(d3Color.r, d3Color.g, d3Color.b) ? 'white' : 'black';
```

Comment:
Could you please use a css class instead of enforcing the color via js?
this will allows you to reuse the same color variable used on the axis labels and on the axis title `$visTextColor` that is theme dependant. means this color will be dark gray if the theme is the light one, and will be white if the theme is the dark theme.

I think you can define a scss variable like if I'm not wrong:
```
$visBarLabelDarkColor: $euiColorDarkShade;
$visBarLabelLightColor: $euiColorLightShade;
```
and apply them through a class:
```
.bar-label--dark {
 fill: $visBarLabelDarkColok;
}
.bar-label--light {
 fill: $visBarLabelLightColok;
}
```

Reply:
Good idea for implementation, but please notice that euiColorDarkShade & euiColorLightShade is not really black and white. In some background colors, it may appear with less contrast.  Here are a screenshot:
![image](https://user-images.githubusercontent.com/8614638/59712492-ff7f1d80-9215-11e9-8087-790365484575.png)

Reply:
I'll do this change, so it will later easy for the team to tweak the css to desired set of colors.

Reply:
FWIW, EUI has variables for full black and white as well: `euiColorFullShade` and `euiColorEmptyShade `, respectively.

Reply:
@ryankeairns those values actually change based on the theme so Full -> becomes white in dark theme. The background colors don't change based on theme so the text colors need to stay consistently white or black. @mmeshi is now using CSS and has already changed these to use _some_ EUI variables: https://github.com/elastic/kibana/blob/c076009251f07c887e070eba0d57da55744b4158/src/legacy/ui/public/vislib/visualizations/point_series/_labels.scss#L1-L2

Reply:
Ah yeah, always confuse those and ghost. (And soon ink!)